### PR TITLE
Make the signal handler exit immediately

### LINKED ISSUE #22240: Crash handler invoking intrinsics::abort sometimes causes recursive crash handling to occur
This is caused by an interaction with the WasmFaultHandler from mozjs that I don't quite understand yet. It's not clear why this would only affect us some of the time.

### PR DESCRIPTION
**Pull request by jdm**
This addresses #22240 by immediately exiting, rather than sometimes triggering a SIGBUS by aborting that can be caught recursively.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #22240
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22259)

### CODE REVIEW

**Comment by CYBAI** [comment]
```
error[E0432]: unresolved imports `libc::pid_t`, `libc::kill`
 --> C:\buildbot\.cargo\registry\src\github.com-1ecc6299db9ec823\sig-1.0.0\src\ffi.rs:8:16
  |
8 | pub use libc::{pid_t, c_int, getpid, kill};
  |                ^^^^^                 ^^^^ no `kill` in the root
  |                |
  |                no `pid_t` in the root

error[E0432]: unresolved imports `libc::sigaction`, `libc::sighandler_t`, `libc::sigfillset`
  --> C:\buildbot\.cargo\registry\src\github.com-1ecc6299db9ec823\sig-1.0.0\src\lib.rs:27:12
   |
27 | use libc::{sigaction, sighandler_t, sigfillset};
   |            ^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^ no `sigfillset` in the root
   |            |          |
   |            |          no `sighandler_t` in the root
   |            no `sigaction` in the root

error: aborting due to 2 previous errors
```

**Comment by CYBAI** [comment]
In `linux-rel-wpt`:

```
error: unused `#[macro_use]` import
 --> ports/servo/non_android_main.rs:6:30
  |
6 | #[cfg(feature = "unstable")] #[macro_use] extern crate sig;
  |                              ^^^^^^^^^^^^
  |
  = note: `-D unused-imports` implied by `-D warnings`

error: unused import: `libc::_exit`
  --> ports/servo/non_android_main.rs:16:5
   |
16 | use libc::_exit;
   |     ^^^^^^^^^^^

error: aborting due to 2 previous errors
```

In `windows-msvc-dev`:

```
error[E0463]: can't find crate for `sig`
 --> ports\servo\non_android_main.rs:6:43
  |
6 | #[cfg(feature = "unstable")] #[macro_use] extern crate sig;
  |                                           ^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error
```

**Comment by jdm** [comment]
@bors-servo retry
* #22276

**Comment by jdm** [comment]
@bors-servo retry
* #22276

**Comment by jdm** [comment]
I begin to suspect that these changes are not as innocuous as they would otherwise appear.

File: ports/servo/non_android_main.rs [hunk 1]
```diff
@@ -3,7 +3,8 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 #[macro_use] extern crate lazy_static;
-#[cfg(feature = "unstable")] #[macro_use] extern crate sig;
+#[cfg(all(feature = "unstable", any(target_os = "macos", target_os = "linunx")))]
```
Comment (paulrouget) [inline ports/servo/non_android_main.rs]: s/linunx/linux

**Comment by paulrouget** [comment]
`target_os = "linunx"`  instead of `target_os = "linux"`? (3 times)

**Comment by CYBAI** [comment]
> 💔 Test failed - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/11555)

Failed at `shell__5 'bash ./etc/ci/lockfile_changed.sh' failed `

```diff
diff --git a/Cargo.lock b/Cargo.lock
index 600c3cd..58f8b71 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3640,7 +3640,7 @@ dependencies = [
  "glutin 0.19.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "keyboard-types 0.4.4 (registry+https://github.com/rust-lang/crates.io-index)",
  "lazy_static 1.2.0 (registry+https://github.com/rust-lang/crates.io-index)",
- "libc 0.2.42 (registry+https://github.com/rust-lang/crates.io-index)",
+ "libc 0.2.44 (registry+https://github.com/rust-lang/crates.io-index)",
  "libservo 0.0.1",
  "log 0.4.5 (registry+https://github.com/rust-lang/crates.io-index)",
  "osmesa-src 0.1.0 (git+https://github.com/servo/osmesa-src)",
@@ -3924,7 +3924,7 @@ name = "sig"
 version = "1.0.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 dependencies = [
- "libc 0.2.42 (registry+https://github.com/rust-lang/crates.io-index)",
+ "libc 0.2.44 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 ```

**Comment by jdm** [comment]
@bors-servo r+
The dangers of rebasing old code.

**Pull request merged**